iothub/device/samples/convention-based-samples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -129,8 +129,13 @@ private async Task HandleTargetTemperatureUpdateRequestAsync(string componentNam
             _logger.LogDebug($"Property: Received - component=\"{componentName}\", [ \"{targetTemperatureProperty}\": {targetTemperature}°C ].");
 
             _temperature[componentName] = targetTemperature;
+            IWritablePropertyResponse writableResponse = _deviceClient
+                .PayloadConvention
+                .PayloadSerializer
+                .CreateWritablePropertyResponse(_temperature[componentName], StatusCodes.OK, version, "Successfully updated target temperature.");
+
             var reportedProperty = new ClientPropertyCollection();
-            reportedProperty.Add(componentName, targetTemperatureProperty, _temperature[componentName], StatusCodes.Accepted, version, "Successfully updated target temperature.");
+            reportedProperty.AddComponentProperty(componentName, targetTemperatureProperty, writableResponse);
```

Comment:
The user is recommended to create an instance of `IWritableProperty` response (we've added this to doc comments). This will also tie well with how they retrieve this value using the `TryGetValue` APIs, as this will give then the entire reponse object directly, instead of giving them the property value, ack code, ack version and ack description as separate values.